src/internal/connector/onedrive/collection.go

```diff
@@ -253,6 +255,9 @@ func (oc *Collection) populateItems(ctx context.Context) {
 			}
 
 			itemReader := lazy.NewLazyReadCloser(func() (io.ReadCloser, error) {
```

Comment:
Thoughts on moving this up to line 211 instead?  That would treat the item download as part of the lazy read.  The only performance gain we achieve by separating the lazy reader and the item download is the pre-download for the first 5 items, and then we're bound to kopia multi-threading.

Reply:
+1

Reply:
We discussed a bit offline - we also get some error handling for the URL i.e. this helps validate the URL is actually valid. Follow-up from the discussion was that we will need to handle the case where the URL might expire for long-running operations - so it might make sense to do what you're suggesting (defer the reader creation and GET to when we start downloading the file and also add some revalidation logic in the reader)